src/release-9.md

```diff
@@ -0,0 +1,236 @@
+is expensive to recompute. In future, systems will be enhanced to recognize and
+exploit features such as one way coupling of matrix blocks. The old functions
+such as `solve_linear` and `newton` will be deprecated in the next development
+cycle for removal in Nutils 11. All examples are updated to use `System`
+directly to encourage new projects to switch over to the new API.
+
+### Field
+
+Nutils 8 initiated the transition from operating on basis functions and vector
+valued residuals, to fields, i.e. bases contracted with arguments, and scalar
+valued residuals. Nutils 9 further promotes this by providing a new `field`
+method that makes formation of the basis an entirely internal affair. In the
+following example, `u1`, `u2` and `u3` are identical:
+
+```python
+>>> u1 = topo.field('u', btype='spline', degree=2)
+>>> basis = topo.basis('spline', degree=2)
+>>> u2 = basis @ function.Argument('u', basis.shape)
+>>> u3 = function.field('u', basis)
+```
+
+In the majority of cases that direct acces to the basis is not required, the
+first construction is recommended. The second shows what really happens,
+discounting some subtle differences in the case of tensorial topologies. The
+third form showcases the new function `function.field`, which was renamed from
+`function.dotarg` for naming consistency. Like its predecessor, this function
+can also be used to create constant fields by not providing a basis argument.
+
+### Factor
+
+Another standout new feature is `function.factor`, which offers to speed up
+calculations by precomputing assembly loops. Its use is limited to functions
+that are already bound to all spaces, e.g. by integration, and that are
+polynomial in all their arguments. Building on the previous example, a typical
+use would be to factor the system argument:
+
+```python
+>>> sys = solver.System(function.factor(residual), trial='u', test='v')
+```
+
+Here, residual must by polynomial in 'u' and 'v'. Factor then evaluates the
+polynomial coefficients as sparse arrays. Finally, when System evaluates the
+resulting function, this amounts to nothing more than sparse tensor
+contractions, which can be very fast. Note, however, a high polynomial degree
+results in high-dimensional coefficients which, while sparse, may still require
+a lot of memory and many operations to perform the required contractions. As
+such, the benefits of factor should be assessed on a case by case basis.
+
+Two examples that benefit greatly from the factor treatment are cylinderflow
+and cahnhilliard. In another useful application of factor, these scripts have
+greatly sped up post processing by factoring sampled results. The `Sample.bind`
+method used for this purpose is also new in Nutils 9.
+
+### JIT compiler
+
+Under the hood, a major change in the evaluable module is the way that
+functions are evaluated. Where evaluations formerly consisted of a (dynamically
+created) sequence of function calls, Nutils 9 purpose generates and compiles a
```

Comment:
purpose generates?

Reply:
changed to generates and compiles a dedicated function